prb2/process_monitor_mate.sh

```diff
@@ -16,7 +15,7 @@ monitor_process() {
             echo "The process has crashed. Restarting..."
             start_process "$@"
         fi
-        sleep 1
+        sleep 10
```

Comment:
you are waiting for 10 sec but in that 10 sec if same PID will be assigned to some other Process in that case you can't be able to find out that your process has been stopped or crashed.